packages/testsuite/cypress/e2e/undertow/test-configuration-subsystem-undertow-ajp-listener.cy.ts

```diff
@@ -0,0 +1,66 @@
+describe("TESTS: Configuration => Subsystem => Undertow => Global settings", () => {
+  let managementEndpoint: string;
+
+  const testValues = {
+    serverName: "default-server",
+    ajpListener: "test-ajp",
+    ajpSocketBindings: "ajp",
+    allowedReqAttrsPatternExpression: "${NON-EXISTING:value.*}",
+    allowedReqAttrsPatternResolved: "value.*",
+  };
+  const address = ["subsystem", "undertow", "server", testValues.serverName, "ajp-listener", testValues.ajpListener];
+  const serverSelectors = {
+    serverListenersItem: "#undertow-server-listener-item",
+    ajpListenerItem: "#undertow-server-ajp-listener-item",
+  };
+  const ajpListenerPageSelectors = {
+    ajpListenersTableId: "undertow-server-ajp-listener-table",
+  };
+  const ajpListenerForm = {
+    id: "undertow-server-ajp-listener-form",
+    allowedReqAttrsPattern: "allowed-request-attributes-pattern",
+  };
+
+  before(() => {
+    cy.startWildflyContainer()
```

Comment:
Yes, you could define optional additional start parameter to: https://github.com/istraka/berg/blob/EAPQE-2410/packages/testsuite/cypress.config.ts#L19

and here just use it (`additionalStartParameter="--stability=preview"`) 

what do you think @OndrejKotek @istraka?

Reply:
@OndrejKotek added. I didn't add the parameter to the task, because I assume I would have to modify tests (setting the parameter). Now I control it via environment variable.

Reply:
@istraka i don't like this approach for two reasons. You have to edit job automation and you don't see the test require a preview mode. could you please introduce a new parameter which you will define when the server is stared. 

You find inspiration in hostMode implemented for OIDC. This is almost same case.

https://github.com/hal/berg/blob/main/packages/testsuite/cypress/e2e/elytron-oidc-client/test-oidc-security-configure-oidc.cy.ts#L62

Reply:
My question is: do we want to have tests in the TS that runs in preview mode? And what if they are promoted to higher stability level? Should the test change?

Reply:
> IIUC this solution would always start wildfly in preview stability mode

I don't think. You can define the default value to false and only in your case you can set it to true. so only your test you will start server in preview mode.

Like in case hostMode. Default value is network mode so when you will not use any parameter it will started in network mode. And only OIDC test define the parameter `useNetworkHostMode: true` so only when the parameter is present then the server will start a server in hostMode. 

> However we want to test the feature NOT in preview mode (in the future)

When this will be changed there have to be some JIRA to track this change. Because the documentation have to be also update etc. So also these test have to update to remove the parameter for preview mode. 

Now you still adding the required subsystem manually and when it will be promoted to regular feature you will have to remove this manual configuration. 

You will have to update a tests in every case so why not to do property and explicitly define the preview mode is active.  

> do we want to have tests in the TS that runs in preview mode? And what if they are promoted to higher stability level? Should the test change?

No we wont start server in preview mode only for your test. 

> And what if they are promoted to higher stability level? Should the test change?

Only the parameter from your test will be removed.

Reply:
I am afraid of forgotten test running in preview stability mode. We have discussed this and agreed we should not do that and I'd like to wait with this PR for WFLY-19808.

We are waiting for the feature to be promoted to the default stability level WFLY-19808. EAP7-1836 is blocked by EAPDOC-2314.

IIUC we need the RFE to be in a sprint for WFLY-19808. But Bartosz has submitted the MR, and if it gets merged....well, we can merge this.